src/sonic-dhcp-server/dhcp_server/common/dhcp_db_monitor.py

```diff
@@ -90,42 +125,128 @@ def _check_dhcp_server_update(self, enabled_dhcp_interfaces):
             if op == "DEL":
                 if key in enabled_dhcp_interfaces:
                     need_refresh = True
-        return need_refresh
+            if need_refresh:
+                self.db_event_executor.pop_db_update_event(self.subscribe_dhcp_server_table)
+                return True
+        return False
+
+
+class DhcpServdDbMonitor(object):
+    subscribe_dhcp_server_table = None
+    subscribe_dhcp_server_port_table = None
+    subscribe_dhcp_server_range_table = None
+    subscribe_vlan_table = None
+    subscribe_vlan_member_table = None
+    subscribe_vlan_intf_table = None
+    subscribe_dhcp_server_options_table = None
 
-    def _check_vlan_update(self, enabled_dhcp_interfaces):
+    def __init__(self, db_connector, select_timeout=DEFAULT_SELECT_TIMEOUT):
+        self.db_connector = db_connector
+        self.sel = swsscommon.Select()
+        self.select_timeout = select_timeout
+        self.db_event_executor = _DbEventExecutor()
+
+    def subscribe_table(self):
         """
-        Check vlan table
-        Args:
-            enabled_dhcp_interfaces: DHCP interface that enabled dhcp_server
-        Returns:
-            Whether need to refresh
+        Subcribe db table to monitor
         """
-        need_refresh = False
-        while self.subscribe_vlan_table.hasData():
-            key, op, _ = self.subscribe_vlan_table.pop()
-            # For vlan doesn't have related dhcp entry, not need to refresh dhcrelay process
-            if key not in enabled_dhcp_interfaces:
-                continue
-            need_refresh = True
-        return need_refresh
+        self.subscribe_dhcp_server_table = swsscommon.SubscriberStateTable(self.db_connector.config_db,
```

Comment:
how about use a dict to store subscribed tables?

self.subscribed_tables[xxx] = xxx

Reply:
Fixed